c2rust-refactor/src/scripting/to_lua_ast_node.rs

```diff
@@ -286,15 +450,258 @@ unsafe impl Send for LuaAstNode<P<Expr>> {}
+                e => unimplemented!("LuaAstNode<P<Expr>>:set_ty() for {}", e.ast_name()),
+            }
+
+            Ok(())
+        });
+
+        methods.add_method("get_ident", |lua_ctx, this, ()| {
+            match &this.borrow().node {
+                ExprKind::Field(_, ident) => ident.to_lua(lua_ctx).map(|i| Some(i)),
+                _ => Ok(None),
+            }
+        });
+
+        methods.add_method("get_path", |_lua_ctx, this, ()| {
+            match &this.borrow().node {
+                ExprKind::Path(_, path)
+                | ExprKind::Struct(path, ..) => Ok(LuaAstNode::new(path.clone())),
+                e => unimplemented!("LuaAstNode<P<Expr>>:get_path() for {}", e.ast_name()),
+            }
+        });
+
+        methods.add_method("get_exprs", |_lua_ctx, this, ()| {
+            match &this.borrow().node {
+                ExprKind::Cast(expr, _)
+                | ExprKind::Field(expr, _)
+                | ExprKind::Unary(_, expr) => Ok(vec![LuaAstNode::new(expr.clone())]),
+                ExprKind::MethodCall(_, exprs) => Ok(exprs.iter().map(|e| LuaAstNode::new(e.clone())).collect()),
+                ExprKind::Assign(lhs, rhs) => Ok(vec![LuaAstNode::new(lhs.clone()), LuaAstNode::new(rhs.clone())]),
+                ExprKind::Call(func, params) => {
+                    let mut exprs = Vec::with_capacity(params.len() + 1);
+
+                    exprs.push(LuaAstNode::new(func.clone()));
+
+                    for param in params {
+                        exprs.push(LuaAstNode::new(param.clone()));
+                    }
+
+                    Ok(exprs)
+                },
+                _ => Ok(Vec::new()),
+            }
+        });
+
+        methods.add_method("set_exprs", |_lua_ctx, this, exprs: Vec<LuaAstNode<P<Expr>>>| {
+            match &mut this.borrow_mut().node {
+                ExprKind::Field(expr, _)
+                | ExprKind::Unary(_, expr) => *expr = exprs[0].borrow().clone(),
+                ExprKind::MethodCall(_, exprs) => *exprs = exprs.iter().map(|e| e.clone()).collect(),
+                ExprKind::Call(func, params) => {
+                    *func = exprs[0].borrow().clone();
+                    *params = exprs.iter().skip(1).map(|e| e.borrow().clone()).collect()
+                },
+                ExprKind::Assign(lhs, rhs)
+                | ExprKind::Binary(_, lhs, rhs) => {
+                    *lhs = exprs[0].borrow().clone();
+                    *rhs = exprs[1].borrow().clone();
+                },
+                ExprKind::Cast(expr, _) => *expr = exprs[0].borrow().clone(),
+                e => unimplemented!("LuaAstNode<P<Expr>>:set_exprs() for {}", e.ast_name()),
+            }
+
+            Ok(())
+        });
+
+        methods.add_method("get_op", |_lua_ctx, this, ()| {
+            match &this.borrow().node {
+                ExprKind::Unary(op, _) => Ok(Some(op.ast_name())),
+                // ExprKind::Binary(op, ..) |
```

Comment:
I think I didn't need them implemented, but acknowledged that they probably belong there too. I think `BinOp` didn't have `AstName` implemented for some reason, so I didn't go through the trouble of manually adding it since I didn't need it. (But weirdly `UnOp` had `AstName` impl'd already)

Reply:
Sounds good, just leave a TODO